hibernate-core/src/main/java/org/hibernate/hql/internal/ast/util/SyntheticAndFactory.java

```diff
@@ -147,7 +147,7 @@ public void addDiscriminatorWhereFragment(
 			Map enabledFilters,
 			String alias) {
 		String whereFragment = persister.filterFragment( alias, enabledFilters ).trim();
-		if ( "".equals( whereFragment ) ) {
+		if ( whereFragment.isEmpty() ) {
```

Comment:
Not null safe anymore but I suppose it can't happen considering the line before.